pkg/nsx/services/compare.go

```diff
@@ -2,93 +2,109 @@ package services
 
 import (
 	"encoding/json"
-	"sort"
 
 	"github.com/vmware/vsphere-automation-sdk-go/runtime/data/serializers/cleanjson"
 	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
 )
 
-func SecurityPolicyEqual(existingSecurityPolicy *model.SecurityPolicy, securityPolicy *model.SecurityPolicy) bool {
-	s1, _ := json.Marshal(simplifySecurityPolicy(existingSecurityPolicy))
-	s2, _ := json.Marshal(simplifySecurityPolicy(securityPolicy))
-	if string(s1) == string(s2) {
-		return true
+func (service *SecurityPolicyService) securityPolicyEqual(sp1 *model.SecurityPolicy, sp2 *model.SecurityPolicy) *model.SecurityPolicy {
+	v1, _ := json.Marshal(service.simplifySecurityPolicy(sp1))
+	v2, _ := json.Marshal(service.simplifySecurityPolicy(sp2))
+	if string(v1) == string(v2) {
+		return nil
 	}
-	log.Info("security policy diff", "nsx sp", s1, "k8s sp", s2)
-	return false
+	log.V(1).Info("security policies differ", "current NSX security policy", service.simplifySecurityPolicy(sp1),
+		"desired NSX security policy", service.simplifySecurityPolicy(sp2))
+	return sp2
 }
 
-func RulesEqual(existingRules []model.Rule, rules []model.Rule) (bool, []model.Rule) {
-	// sort the rules by id, otherwise expandRule may return different results, only the sequence of the
-	// rule is different, so sort by port number, and it avoids the needless updates.
-	var sortRules = func(rules []model.Rule) {
-		sort.Slice(rules, func(i, j int) bool {
-			return *(rules[i].Id) > *(rules[j].Id)
-		})
-	}
-	sortRules(existingRules)
-	sortRules(rules)
+func (service *SecurityPolicyService) rulesEqual(existingRules []model.Rule, rules []model.Rule) ([]model.Rule, []model.Rule) {
```

Comment:
Can we rename `rulesEqual` and `groupsEqual` to `rulesCompare` and `groupsCompare`? Because I feel that the function xxxEqual should return a value of boolean type...

Reply:
ok